src/dynamic_compiler.c

```diff
@@ -2398,17 +2398,28 @@ char *dynamic_compile_prepare(char *fld0, char *fld1) {
 }
 char *dynamic_compile_split(char *ct) {
 	extern int ldr_in_pot;
+	static char Buf[1024], Buf2[1024];
 	if (strncmp(ct, "dynamic_", 8)) {
-		return dynamic_compile_prepare("", ct);
+		char *cp;
+		ct = dynamic_compile_prepare("", ct);
+		cp = strrchr(ct, '@');
+		while (*cp) {
+			if (*cp >= 'A' && *cp <= 'Z') {
+				snprintf(Buf, sizeof(Buf), "%s", ct);
+				cp = strrchr(Buf, '@');
+				strlwr(cp);
+				return Buf;
+			}
+			++cp;
+		}
+		return ct;
```

Comment:
What is the above code doing? Will it not lower-case constants in the expression?

Reply:
You are right, but not constants.   The only thing it will screw up (I think) is salts.  Constants are p[art of the format, not the input hashes.  I could add a check for a '$' before the check for upper case, and if I find the '$', simply return ct.

I also see that Buf2 (earlier code) is not being used, and needs removed.

Reply:
Looking deeper (thanks for the poke), there is about no way to do this, and this code will have to be reverted.  It would kill things like:

```--format='MD5($p)'````

Also fixing valid will be equally as hard (for same reasons), !!!   damn.  

Looks like I will have to keep the code in the case-test-3 as a ```--test-full=1``` level, since I am now not sure I can get dyna expression format to pass that test.